torch/csrc/jit/script/sugared_value.h

```diff
@@ -301,13 +299,19 @@ struct TORCH_API CastValue : public BuiltinFunction {
   TypePtr type_;
 };
 
+using SugaredValuePtr = std::shared_ptr<SugaredValue>;
+
 // builtins operators and functions that call a method if it exists
 // on a class type, like 'len(x)' and 'x + y'
-struct TORCH_API OperatorOverload : public BuiltinFunction {
-  OperatorOverload(c10::Symbol builtin_method, std::string desugared_name)
-      : BuiltinFunction(builtin_method, c10::nullopt),
+struct TORCH_API MagicMethod : public SugaredValue {
```

Comment:
If this is coupled to `CastValue`, why not combine them so we just have 1 sugared value for builtins?